### PR TITLE
Simulation from the original VATA library

### PR DESCRIPTION
**Pull request by martinhruska**
This pull request:

-  Imports computation of simulation from the original VATA library
-  Imports the related classes
-  Is based on migration of simulation to the previous library by Juraj.
-  Implements tests for simulation
-  Fixes issues reported by clang-tide and cppcheck
-  Changes names of functions from camel style to snake style

### CODE REVIEW

**Comment by ondrik** [comment]
I don't know, what to check. Is it working?

Maybe some comments to add, based on some recent thinking about simulation:
1. importing `#include <vata2/explicit_lts.hh>` in `nfa.hh` is strange, the header should be imported only from some `.cc` file that needs it
2. is it good to have a function named `compute_simulation(const NFA&)`? I'd at least add a `StringDict` parameter, so that one can choose what simulation they want to compute (preferably also do not mention 'simulation' in the name... what if someone wants to compute bisimulation? should we introduce another name?): such as "forward", "backward", "lookahead", or some under-approximations. Better keep the interface small and move the dispatching into some `.cc` file
3. I think it might make sense to separate the simulations into their own (small) library (maybe not now, but in not so far future). I wish to play with some simulation under-approximations and preprocessing to be used in SPOT, and it would be good if it could be used also here without too much pain.

**Comment by martinhruska** [comment]
The points 1,2 are solved, an issue was created for the point 3. Now, we wait for review by @tfiedor.

File: include/vata2/simutil/binary_relation.hh
Comment (tfiedor) [inline include/vata2/simutil/binary_relation.hh:78]: Dead code, is it necessary?

File: include/vata2/simutil/binary_relation.hh
Comment (tfiedor) [inline include/vata2/simutil/binary_relation.hh:186]: Dead code.

File: include/vata2/simutil/binary_relation.hh
Comment (tfiedor) [inline include/vata2/simutil/binary_relation.hh:208]: Pure gem :D

File: include/vata2/simutil/binary_relation.hh
Comment (tfiedor) [inline include/vata2/simutil/binary_relation.hh:215]: Suggestion: rename to `build_equivalence_classes`

File: include/vata2/simutil/binary_relation.hh
Comment (tfiedor) [inline include/vata2/simutil/binary_relation.hh:391]: Can't say I'm advocate for snake case of other case, but I would probably pick one and be consisent through the project.

File: src/nfa/nfa.cc
Comment (tfiedor) [inline src/nfa/nfa.cc:890]: Suggestion: add default to the params `{'relation', 'simulation'}` or whatever it is called. (See other `StringDict& params` usages.)

**Review by tfiedor — APPROVED** [review]
Nothing major from me. There is some dead code, also I noticed some inconsistencies in naming (since we use snake case, I would use snakecase).

The only thing that I would definitely change is the default parameter in the computing simulation, as it is better for generic usage to provide some default that users can use without additional pain.

File: include/vata2/simutil/binary_relation.hh
Comment (martinhruska) [reply include/vata2/simutil/binary_relation.hh]: removed.

File: include/vata2/simutil/binary_relation.hh
Comment (martinhruska) [reply include/vata2/simutil/binary_relation.hh]: removed.

File: include/vata2/simutil/binary_relation.hh
Comment (martinhruska) [reply include/vata2/simutil/binary_relation.hh]: solved

File: include/vata2/simutil/binary_relation.hh
Comment (martinhruska) [reply include/vata2/simutil/binary_relation.hh:192]: yes, this is a gem and it should be left here for future generations as a documentation of Ondra's feelings when he studied Jiri's code.

**Pull request merged**
